dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventReferralValidationHook.java

```diff
@@ -0,0 +1,80 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.validation.hooks;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.hisp.dhis.tracker.report.TrackerErrorCode.E1311;
+import static org.hisp.dhis.tracker.report.TrackerErrorCode.E1312;
+
+import java.util.List;
+
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.domain.Relationship;
+import org.hisp.dhis.tracker.report.ValidationErrorReporter;
+import org.hisp.dhis.util.ObjectUtils;
+import org.springframework.stereotype.Component;
+
+/**
+ *
+ * Validation hook to make sure referral events contain Relationship and both
+ * sides of Relationship are provided
+ *
+ * @author Zubair Asghar
+ */
+
+@Component
+public class EventReferralValidationHook extends AbstractTrackerDtoValidationHook
+{
+    @Override
+    public void validateEvent( ValidationErrorReporter reporter, TrackerBundle bundle, Event event )
+    {
+        ProgramStage programStage = bundle.getPreheat().getProgramStage( event.getProgramStage() );
+
+        checkNotNull( programStage, TrackerImporterAssertErrors.PROGRAM_STAGE_CANT_BE_NULL );
+
+        validateReferralRelationshipLinkage( reporter, event, programStage );
+    }
+
+    public void validateReferralRelationshipLinkage( ValidationErrorReporter reporter,
+        Event event, ProgramStage programStage )
+    {
+        if ( programStage.isReferral() )
+        {
+            List<Relationship> relationships = event.getRelationships();
+
+            reporter.addErrorIf( relationships::isEmpty, event, E1311,
+                event.getEvent() );
```

Comment:
now that the error message has no argument anymore you should also remove the arg event.getEvent() here https://github.com/dhis2/dhis2-core/pull/10986/files#diff-5bda4b7c88a18af614d4dfde1ba2055b19e4ee287b096db8c53e1491f3cb077aR73 and here https://github.com/dhis2/dhis2-core/pull/10986/files#diff-5bda4b7c88a18af614d4dfde1ba2055b19e4ee287b096db8c53e1491f3cb077aR77